### DISCUSSION TITLE
multiple individual time series

### DISCUSSION THREAD

**Discussion by khalidbouziane** [pycaret/pycaret#2690] [Q&A]
Sometimes a dataset can contain multiple individual time series trends, where a series identifier column indicates which series each row belongs to. This series ID is part of the input data, and the model can use it to segment the series and learn each individually. If there are relationships or commonalities between the series, the model can ideally take advantage of those relationships.

Example: given individual time series trends across 100 stores, predict next week's sales, for any of those stores.

when I tried to train the type of data mentioned above on pycaret ts, I faced the issue of setting up the index in setup function, as the index must not have duplicate values, but in my case since an event can happen multiple times in the same day, or week or month, for example different customer can make a purchase in the same month (index) ,and I want to predict  next month purchase for any of those customer

Is there a way to play around this ?

**Comment by Beenjamming** [comment]
My understanding is that this is not currently possible in the current version.  

Moez had posted this previously using the regression model:

https://towardsdatascience.com/multiple-time-series-forecasting-with-pycaret-bc0a779a22fe

I have been trying to figure out a way to do this also.  Would love to be able to apply the TS module in this way.
